qmerge.c

```diff
@@ -919,12 +919,17 @@ pkg_merge(int level, const depend_atom *atom, const struct pkg_t *pkg)
 	 * name suggests, bug #660508, usage of BINPKG_COMPRESS,
 	 * due to the minimal nature of where we run, we cannot rely on file
 	 * or GNU tar, so have to do some laymans MAGIC hunting ourselves */
-	compr = "j";  /* default: bzip2 */
+	compr = "I brotli"; /* default: brotli; has no magic header */
 	{
-		/* bz2: 2-byte: 'B' 'Z'                  at byte 0
-		 * gz:  4-byte:  1f  8b                  at byte 0
+		/* bz2: 3-byte: 'B' 'Z' 'h'              at byte 0
+		 * gz:  2-byte:  1f  8b                  at byte 0
```

Comment:
2 byte and 4 byte don't match here

Reply:
I don't understand what you mean. The "magic" string for bz2 is 3 bytes long, the one for gz is 2 bytes long.

Reply:
1f 8b is 4 bytes.

Reply:
Hu? "1f" is one byte and "8b" is a second byte.
In the previous test you indeed tested 4 bytes: 1f 8b 08 \0
But I think these bytes 3 and 4 need not necessarily have these values. (At least, neither portage nor the "file" utility do expect these value, and according to http://www.onicos.com/staff/iz/formats/gzip.html some bits in bytes 3 and 4 are marked as "reserved" or have special meaning.)

Reply:
never mind me, I was wrong and confused